Add (Go) Logrus documentation (#10974)
getsentry/sentry-docs: docs/platforms/go/guides/logrus/index.mdx

---
title: Logrus
description: "Logrus is a structured logger for Go, used to log messages in different formats and levels. This guide demonstrates how to integrate Logrus with Sentry to capture and send logs to Sentry."
---

For a quick reference, there is a [complete example](https://github.com/getsentry/sentry-go/tree/master/_examples/logrus) at the Go SDK source code repository.

> [Go Dev-style API documentation](https://pkg.go.dev/github.com/getsentry/sentry-go/logrus) is also available.

Reviewer: ```suggestion
Go API documentation for the [`sentrylogrus` package](https://pkg.go.dev/github.com/getsentry/sentry-go/logrus) is also available.
```
Original link text was vague and sounded less relevant. I think something like this would make me more likely to click the link. Feel free to adjust the text.

	// Initialize Sentry
	sentryHook, err := sentrylogrus.New(sentryLevels, sentry.ClientOptions{
		Dsn: "___PUBLIC_DSN___",
		BeforeSend: func(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
			if hint.Context != nil {
				if req, ok := hint.Context.Value(sentry.RequestContextKey).(*http.Request); ok {
					// Access the original Request here
					fmt.Println(req)
				}
			}
			fmt.Println(event)
			return event
		},
		Debug:            true,
		AttachStacktrace: true,
	})
	if err != nil {
		panic(err)
	}
	defer sentryHook.Flush(5 * time.Second)
	logger.AddHook(sentryHook)

	// Flushes before calling os.Exit(1) when using logger.Fatal
	// (else all defers are not called, and Sentry does not have time to send the event)
	logrus.RegisterExitHandler(func() { sentryHook.Flush(5 * time.Second) })

	// Log an InfoLevel entry to STDERR (not sent to Sentry)
	logger.Infof("Application has started")

	// Log an ErrorLevel entry to STDERR and Sentry
	logger.Errorf("oh no!")

	// Log a FatalLevel entry to STDERR, send to Sentry, and terminate the application
	logger.Fatalf("can't continue...")
    
```

## Configure

> `sentrylogrus` allows configuration via the New function, which accepts the levels to log and sentry.ClientOptions. The levels to log are the logrus levels that should be sent to Sentry. The `sentry.ClientOptions` are the same as the ones used in the `sentry.Init` function.

Reviewer: ```suggestion
`sentrylogrus` allows configuration via the `New` function, which accepts the levels to log and `sentry.ClientOptions`. The levels to log are the logrus levels that should be sent to Sentry. The `sentry.ClientOptions` are the same as the ones used in the `sentry.Init` function.
```
